source/settings.rst

```diff
@@ -215,25 +215,62 @@ depending on whether or not they are distributed with that repository.
 
     .. settingsgroup:: Author images
       :id: author-images
+      
+      .. setting:: Show author's avatar column in the commit graph
+        :id: show-avatar-commit-graph
 
-      .. setting:: Get author image from gravatar.com
-        :id: gravatar
+        If checked, avatar images are downloaded for commit authors and shown on the right side in the commit graph.
 
-        If checked, `gravatar <http://gravatar.com/>`_ will be accessed to
-        retrieve an image for the author of commits. This image is displayed on
-        the ``commit`` tab on the main Revision Graph window.
+      .. setting:: Show author's avatar in the commit info view
+        :id: show-avatar-commit-info
 
-      .. setting:: Image size
+        If checked, avatar images are downloaded for commit authors and shown in the commit info view.
 
-        The display size of the user image.
+      .. setting:: Avatar provider
+        :id: avatar-provider
 
-      .. setting:: Cache images
+        The avatar provider setting determines the source from which avatar images are requested.
 
-        The number of days to elapse before gravatar is checked for any changes to an authors image.
+        - ``Default`` - The default avatar provider loads a user defined avatar images, depending on the email address, from GitHub or Gravatar.
+          If no user defined image could be found, a fallback images is used.
```

Comment:
Describe the algorithm more in detail?

```suggestion
        - ``Default`` - The default avatar provider loads a user defined avatar images, depending on the email address, from GitHub 
          (if this is a GitHub repo and the email is ending with @users.noreply.github.com), 
          otherwise query `gravatar <http://gravatar.com/>`.
          Note that GitHub users may configure the account to use Gravatar too.
          If no user defined image could be found, a fallback images is used.
```

Reply:
I'm not sure if the exact details should be in the user manual, because it would be outdated as soon as someone changes a few things.

There are also quite a few edge cases that might be pretty hard (or unnecessary) do describe. For example the bot account detection for github email addresses and the fact that if a github user decides to use gravatar (in the github settings) we still get the avatar from github because github loads it from gravatar and caches it (and it is also a request less for us if the first one already succeeds)